### PR TITLE
Script-blocking style sheet tests

### PR DESCRIPTION
**Pull request by domfarolino**
Tests for https://github.com/whatwg/html/pull/4031.
Explanation of the tests:
 - link-load-fired-before-scripting-unblocked.html
    - This just tests that scripting is not unblocked until the load event of the element contributing a script-blocking style is fired. TODO: identical test, but for error event
 - stylesheet-not-removed-until-next-stylesheet-loads.html
    - This tests that a style sheet does not become unavailable to scripts until a new one loaded from the same link replaces it

### CODE REVIEW

File: html/semantics/document-metadata/the-link-element/link-load-fired-before-scripting-unblocked.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+var saw_link_onload = false;
+var t = async_test("Check if the stylesheet's load event is fired before the " +
+                   "pending parsing-blocking script is unblocked");
+</script>
+<link href="style.css?pipe=trickle(d3)" rel="stylesheet" id="style_test"
+      onload="t.step(function() { saw_link_onload = true; });"
+      onerror="t.step(function() { assert_unreached('Sheet should load OK'); });"></link>
```
Comment (domenic) [inline html/semantics/document-metadata/the-link-element/link-load-fired-before-scripting-unblocked.html]: You can use `t.unreached_func('Sheet should load OK')` if you want

File: html/semantics/document-metadata/the-link-element/stylesheet-not-removed-until-next-stylesheet-loads.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+var t = async_test("Check that the style sheet loaded by an <link> element " +
+                   "is available to scripts until its successor is completely loaded");
+</script>
+<link href="style.css?pipe=trickle(d3)" rel="stylesheet" id="style_test"
+      onerror="t.step(function() { assert_unreached('Sheet should load OK'); });"></link>
+<script>
+  t.step(function() {
+    assert_true(document.styleSheets.length === 1 &&
+                document.styleSheets[0].href.includes("style.css"),
+                "The style sheet 'style.css' is not available to scripts");
```
Comment (domenic) [inline html/semantics/document-metadata/the-link-element/stylesheet-not-removed-until-next-stylesheet-loads.html]: Generally, the assertion messages should be stated as things that should happen (like in your other tests), not as things that should not happen. This is inconsistent across WPT but especially for code readers I think it's the right thing to do.

File: html/semantics/document-metadata/the-link-element/stylesheet-not-removed-until-next-stylesheet-loads.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
+<link rel="author" title="Dom Farolino" href="mailto:dom@chromium.org">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/links.html#link-type-stylesheet">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+var t = async_test("Check that the style sheet loaded by an <link> element " +
```
Comment (domenic) [inline html/semantics/document-metadata/the-link-element/stylesheet-not-removed-until-next-stylesheet-loads.html]: It's unclear why this is an async_test.

File: html/semantics/document-metadata/the-link-element/stylesheet-not-removed-until-next-stylesheet-loads.html [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+                   "is available to scripts until its successor is completely loaded");
+</script>
+<link href="style.css?pipe=trickle(d3)" rel="stylesheet" id="style_test"
+      onerror="t.step(function() { assert_unreached('Sheet should load OK'); });"></link>
+<script>
+  t.step(function() {
```
Comment (domenic) [inline html/semantics/document-metadata/the-link-element/stylesheet-not-removed-until-next-stylesheet-loads.html]: You might as well join these all into one `t.step()`

**Review by domenic — APPROVED** [review]
LGTM with style suggestions

File: html/semantics/document-metadata/the-link-element/stylesheet-not-removed-until-next-stylesheet-loads.html (hunk 3 above)
Comment (domfarolino) [reply html/semantics/document-metadata/the-link-element/stylesheet-not-removed-until-next-stylesheet-loads.html]: Good point; mostly that was copy paste

**Pull request merged**